web-api/src/storage/elastic/filter.rs

```diff
@@ -249,6 +357,110 @@ enum Clause<'a> {
     MustNot(MustNot<'a>),
 }
 
+fn fully_merge_range(mut clause: Vec<Clause<'_>>) -> Vec<Clause<'_>> {
```

Comment:
:wrench: 

```suggestion
fn merge_ranges_in_and_clause(mut clause: Vec<Clause<'_>>) -> Vec<Clause<'_>> {
```

(name is just an arbitrary proposal not what I want it to be)

This is only correctly usable iff applied to a clauses from a `and` statement, I think it's important for the function name to reflect this. At least for me `fully` doesn't do so at all.